### ISSUE TITLE
USPS timeout is not long enough

### ISSUE THREAD

**Issue by SMTDesigns** [nopSolutions/USPS-plugin-for-nopcommerce#6]
The timeout in USPSComputationMethod.cs is no longer long enough and shipping quotes from USPS are now failing depending on USPS website timing. It still fails ocassionaly  at 45 seconds (I modified my own copy to that value). The parameter should not be hard coded and made available for circumstances as a settable parameter.

  public override void Install()
        {
            //settings
            var settings = new USPSSettings
            {
                Url = USPSShippingDefaults.DEFAULT_URL,
                Username = "123",
                Password = "456",
                ClientTimeout = 60,
                AdditionalHandlingCharge = 0,
                CarrierServicesOfferedDomestic = "",
                CarrierServicesOfferedInternational = ""
            };
            _settingService.SaveSetting(settings);

**Comment by skoshelev** [comment]
Hi @SMTDesigns 

This setting is selected from the average response rate of the USPS server and we see no reason to change it. However, this parameter is not hard-coded and is editable from the administration area from the "All settings (advanced)" page, see the uspssettings.clienttimeout setting

Closed #6

**Comment by SMTDesigns** [comment]
the word "average" is the wrong away to look at this. It should be set to a reasonable maximum time. If the site responds quicker, then there is no delay to the customer. The problem is that there is no obvious retry and the customer gets a message saying that USPS isn't available - not e-commerce friendly... It is good to know there is a settable parameter. It should be clearer that it even exists for the admin. My recommendation is to set it to 60 -90 seconds - why do something that is average that is business fatal when is does timeout so quickly given the "average response time is more likely 30 seconds these days. I have had it timeout about once every five or ten tries. Even though I state on my homepage to try again, customers don't... I usually get an email from those that bother to let me know telling me USPS is unavailable.

By the way, I searched for this settable parameter in my 4.30 installation and it does not exist. I went ahead and created the new setting and set it to 90 seconds - because 60 doesn't work all the time and the default works half the time at least in my geographic area.
-----Original Message-----
